### ISSUE TITLE
Expansion Panel - Remove from DOM when closed/minimized

### ISSUE THREAD

**Issue by Gil-Epshtain** [angular/components#14626]
#### Please describe the feature you would like to request.
I would like the "expansion-panel" be removed from the DOM when minimized/closed and not hidden with CSS style.

#### What is the use-case or motivation for this proposal?
I have a "mat-accordion" with a list of "mat-expansion-panel" - a very long list, between 500 to 1000 items.
Each "mat-expansion-panel" has a child element: "mat-expansion-panel-content". This element is hidden with CSS style (visibility: hidden;) when the mat-expansion-panel is minimized/closed. 
Due to this there is a big performance issue, since the browser need to handle all of these hidden DOM objects.

I would like "mat-expansion-panel-content" will be removed from the DOM when the "mat-expansion-panel" is closed - with *ngIf or any other way and not only hidden. 

#### Is there anything else we should know?

I would like to know if there is any work around this issue, if a framework solution (development) will not be presented soon.

**Comment by crisbeto** [comment]
You could use the `matExpansionPanelContent` to render the content lazily only after one of the panels is opened. Note that this won't remove it again after it is closed. Docs: https://material.angular.io/components/expansion/overview#lazy-rendering.

**Comment by Gil-Epshtain** [comment]
I have developed a work-around for this issue, by adding a isOpen parameter to each "mat-expansion-panel". And put the "mat-expansion-panel-content" in a *ngIf.

```
<mat-accordion>
  <mat-expansion-panel *ngFor="let item of list"
                       (opened)="item.isOpen = true"
                       (closed)="item.isOpen = false">
    <mat-expansion-panel-header>
      <mat-panel-title>
        Self aware panel
      </mat-panel-title>
      <mat-panel-description>
        Currently I am {{item.isOpen ? 'open' : 'closed'}}
      </mat-panel-description>
    </mat-expansion-panel-header>
    <div *ngIf="item.isOpen">
      <p>I'm visible because I am open</p>
    </div>
  </mat-expansion-panel>
</mat-accordion>
```
see: [stackblitz](https://angular-grnnbf.stackblitz.io)
however since this isn't a native solution the animation isn't so smooth.

**Comment by andreElrico** [comment]
I was able to construct a directive that does just that:

https://stackblitz.com/edit/angular-os5cc6?file=src%2Fapp%2Fexpansion-overview-example.ts

+ check the js console
+ there is not changeDetection when expansionPanels are closes

Why is this feature not on board? I assume this improves speed of a CD cycle meaning overall performance. --> Problem with my approach - you will loose state (filled forms etc, because you create new instances all the time)

I tried accessing the ChangeDetectorRef of ExpansionPanelComponent but I had no success. It still checks my "cd-check-components".

Same is true for "material tabs" there should be an option for "what I dont See, please dont check". 

Any input/ ideas?

@crisbeto @jelbourn @kara ?

**Comment by crisbeto** [comment]
Duplicate of https://github.com/angular/components/issues/20680.
